src/jit/lclvars.h

```diff
@@ -0,0 +1,750 @@
+    {
+        assert(varTypeIsFloating(type));
+        lvSetHfaTypeIsFloat(type == TYP_FLOAT);
+    }
+
+#ifndef LEGACY_BACKEND
+    var_types lvaArgType();
+#endif
+
+    PerSsaArray lvPerSsaData;
+
+#ifdef DEBUG
+    // Keep track of the # of SsaNames, for a bounds check.
+    unsigned lvNumSsaNames;
+#endif
+
+    // Returns the address of the per-Ssa data for the given ssaNum (which is required
+    // not to be the SsaConfig::RESERVED_SSA_NUM, which indicates that the variable is
+    // not an SSA variable).
+    LclSsaVarDsc* GetPerSsaData(unsigned ssaNum)
+    {
+        assert(ssaNum != SsaConfig::RESERVED_SSA_NUM);
+        assert(SsaConfig::RESERVED_SSA_NUM == 0);
+        unsigned zeroBased = ssaNum - SsaConfig::UNINIT_SSA_NUM;
+        assert(zeroBased < lvNumSsaNames);
+        return &lvPerSsaData.GetRef(zeroBased);
+    }
+
+#ifdef DEBUG
+public:
+    void PrintVarReg() const
+    {
+        if (isRegPairType(TypeGet()))
+        {
+            printf("%s:%s", getRegName(lvOtherReg), // hi32
+                   getRegName(lvRegNum));           // lo32
+        }
+        else
+        {
+            printf("%s", getRegName(lvRegNum));
+        }
+    }
+#endif // DEBUG
+
+}; // class LclVarDsc
+
+class LclVarTable final
+{
+    unsigned m_count;
+    unsigned m_capacity;
+    LclVarDsc* m_array;
+
+    void Grow(Compiler& compiler, unsigned newCapacity);
+    void GrowIfNecessary(Compiler& compiler, unsigned requiredCapacity);
+
+public:
+    class Iterator final
+    {
+        friend class LclVarTable;
+
+        LclVarDsc* m_lclVarDsc;
+
+        Iterator(LclVarDsc* lclVarDsc)
+            : m_lclVarDsc(lclVarDsc)
+        {
+        }
+
+    public:
+        inline LclVarDsc* operator*()
+        {
+            return m_lclVarDsc;
+        }
+
+        inline LclVarDsc* operator->()
+        {
+            return m_lclVarDsc;
+        }
+
+        inline bool operator==(const Iterator& other) const
+        {
+            return m_lclVarDsc == other.m_lclVarDsc;
+        }
+
+        inline bool operator!=(const Iterator& other) const
+        {
+            return m_lclVarDsc != other.m_lclVarDsc;
+        }
+
+        inline Iterator& operator++()
+        {
+            ++m_lclVarDsc;
+            return *this;
+        }
+    };
+
+    LclVarTable();
+    LclVarTable(Compiler& compiler, unsigned initialSize);
+
+    LclVarTable& operator=(const LclVarTable& other);
+
+    LclVarTable(LclVarTable&& other) = delete;
+    LclVarTable(const LclVarTable& other) = delete;
+
+    unsigned Count() const
+    {
+        return m_count;
+    }
+
+    LclVarDsc& operator[](const unsigned index) const
+    {
+        assert(index < m_count);
+        return m_array[index];
+    }
+
+    Iterator begin() const
+    {
+        return Iterator(&m_array[0]);
+    }
+
+    Iterator end() const
+    {
+        return Iterator(&m_array[m_count]);
+    }
+
+    IteratorPair<Iterator> LclVars(unsigned start, unsigned end) const
```

Comment:
>LclVars [](start = 27, length = 7)

I don't feel strongly about this, and I know that it is common usage to use a plural to imply an iterator, but I prefer a more explicit name, e.g. `LclVarIterator`